jdi-light-vuetify/src/main/java/com/epam/jdi/light/vuetify/elements/common/Icon.java

```diff
@@ -29,130 +21,49 @@
-    }
+public class Icon extends UIBaseElement<IconAssert> implements HasClick, HasLabel, HasCursor,
+    HasAlignment, HasTheme, HasMeasurement, HasColor {
 
-    public static List<Icon> finds(UIBaseElement<?> rootElement, String iconName) {
-        return finds(rootElement.core(), iconName);
-    }
-
-    public static Icon find(UIBaseElement<?> rootElement, String iconName) {
-        return finds(rootElement, iconName).get(0);
-    }
-
-    @JDIAction("Cast '{name}' to icon")
-    public static Icon toIcon(UIElement element) {
-        return new Icon().setCore(Icon.class, element);
-    }
-
-    public static List<Icon> finds(UIElement rootElement, String iconName) {
-        WebList elements = rootElement.finds(".mdi-" + CaseFormat.UPPER_UNDERSCORE.to(CaseFormat.LOWER_HYPHEN, iconName));
-        if (elements.size() == 0) {
-            elements = rootElement.finds(String.format("//*[@d='%s']/parent::*/parent::*", getMdiMap().get(iconName)));
-        }
-        if (elements.size() == 0) {
-            elements = rootElement.finds(String.format("//*[contains(@class, 'v-icon') and text()='%s']", iconName.toLowerCase()));
-        }
-        if (elements.size() == 0) {
-            throw new IllegalStateException("No icon with such name: " + iconName);
-        }
-        return elements.stream().map(Icon::toIcon).collect(Collectors.toList());
-    }
-
-    public static Icon find(UIElement rootElement, String iconName) {
-        return finds(rootElement, iconName).get(0);
-    }
-
-    @JDIAction("Get '{name}' mdi-map")
-    public static BidiMap<String, String> getMdiMap() {
-        if (map == null) {
-            map = new TreeBidiMap<>();
-            Pattern mdiName = Pattern.compile("mdi\\w*");
-            Pattern mdiPath = Pattern.compile("\"(.*)\"");
-            try {
-                new BufferedReader(
-                        new InputStreamReader(
-                                new URL("https://raw.githubusercontent.com/Templarian/MaterialDesign-JS/master/mdi.js")
-                                        .openConnection()
-                                        .getInputStream())
-                ).lines().forEach(line -> {
-                    Matcher mdiNameMatch = mdiName.matcher(line);
-                    Matcher mdiPathMatch = mdiPath.matcher(line);
-                    if (mdiNameMatch.find() && mdiPathMatch.find()) {
-                        String name = mdiNameMatch.group().substring(3).replaceAll("(\\B[A-Z])|(\\B[0-9]+)", "_$1$2").toUpperCase();
-                        map.put(name, mdiPathMatch.group(1));
-                    }
-                });
-            } catch (Exception e) {
-                e.printStackTrace();
-            }
-        }
-        return map;
-    }
-
-    @JDIAction("Get '{name}' mdi name")
+    //TODO: implement SVG icon class, implement name, getNameByType methods, map SVG path to a name, example commit: @cbc6e414c1e9edeb853b57e5a3f7007930d500fe
```

Comment:
This is a TODO comment, just so that we won't forget to implement new functionality in the future that are out of scope of the initial task
SVG icons is an interesting component that we may need to implement seperatlyy
This was discussed with @vklonin based on the requirements from @pnatashap

Reply:
thanks for clarification :)